app/components/SourceSelector.vue.ts

```diff
@@ -56,27 +84,56 @@ export default class SourceSelector extends Vue {
       false;
   }
 
-  handleSort(data: any) {
-    const rootNodes = this.scene.getRootNodes();
-    const nodeToMove = rootNodes[data.change.moved.oldIndex];
-    const destNode = this.scene.getRootNodes()[data.change.moved.newIndex];
+  handleSort(treeNodeToMove: ISlTreeNode<ISceneItemNode>, position: ICursorPosition<TSceneNode>) {
+    const nodeToMove = this.scene.getNode(treeNodeToMove.data.id);
+    const destNode = this.scene.getNode(position.node.data.id);
 
-    if (destNode.getNodeIndex() < nodeToMove.getNodeIndex()) {
+    if (position.placement === 'before') {
       nodeToMove.placeBefore(destNode.id);
-    } else {
+    } else if (position.placement === 'after') {
       nodeToMove.placeAfter(destNode.id);
+    } else if (position.placement === 'inside') {
+      nodeToMove.setParent(destNode.id);
     }
+    nodeToMove.select();
   }
 
-  makeActive(sceneItemId: string, ev: MouseEvent) {
+  startScroll(direction: 'top' | 'bottom') {
+    // const slVueTree = this.$refs.slVueTree;
+    // const treeContaier = this.$refs.treeContainer;
+    // this.scrollIntervalId = window.setInterval(() => {
+    //   if (!slVueTree.draggingNode) {
+    //     this.stopScroll();
+    //     return;
+    //   }
+    //   const deltaScroll = direction === 'top' ? -2 : +2;
+    //   treeContaier.scrollTop += deltaScroll;
+    // }, 500);
+  }
```

Comment:
I take it this is the old scrolls system, or an experiment?

Reply:
Looking at your screenshot I could say I forgot to push the last commit. Now it should be ok. Morgan said that she is busy with some anther task now, so I added a day mode and i'm gonna to add a "add folder" button. Unfortunately, Fontawesome doesn't have standart "add folder" icon, so I have to figure out something with it.